47.md

```diff
@@ -63,6 +63,92 @@ Response:
 }
 ```
 
+Below are the commands that are defined in this NIP.
+
+### Pay Invoice
+
+Request:
+```jsonc
+{
+    "method": "pay_invoice",
+    "params": {
+        "invoice": "lnbc50n1..." // invoice to pay, required
+    }
+}
+```
+
+Response:
+```jsonc
+{
+    "result_type": "pay_invoice",
+    "error": {
+        "code": "UNAUTHORIZED",
+        "message": "human readable error message"
+    },
+    "result": {
+        "preimage": "0123456789abcdef..." // preimage of the payment
+    }
+}
+```
+
+### Make Invoice
+
+Request:
+```jsonc
+{
+    "method": "make_invoice",
+    "params": {
+        "amount": 123, // invoice amount in msats, required
+        "description": "some description", // invoice description, optional
+        "description_hash": "31afdf1..", // invoice description hash, optional
+        "expiry": 3600, // invoice expiry in seconds, optional
+    }
+}
+```
+
+Response:
+```jsonc
+{
+    "result_type": "make_invoice",
+    "error": {
+        "code": "UNAUTHORIZED",
+        "message": "human readable error message"
+    },
+    "result": { // result, object. null in case of error.
+        "invoice": "lnbc50n1...",
+        "payment_hash": "31afdf1.."
+    }
+}
+```
+
+### Lookup Invoice
+
+Request:
+```jsonc
+{
+    "method": "lookup_invoice",
+    "params": {
+        "payment_hash": "31afdf1..", // payment hash of the invoice, one of payment_hash or invoice is required
+        "invoice": "lnbc50n1..." // invoice to lookup
```

Comment:
What is the reason why we can't always just use payment_hash to lookup the invoice?

Reply:
from @bumi 

> for some apps it would also be great to lookup an invoice based on the bolt11 invoice otherwise those apps would need to parse the bolt11 invoice and get the payment hash first (which in some environments is annoying)

you are supposed to be able to use either or

Reply:
Yes, but `make_invoice` returns the payment_hash. And we would only lookup the invoices created using `make_invoice`?

Reply:
Possible you could have an application that doesn't call `make_invoice`